package/libyui-ncurses-pkg.spec

```diff
@@ -17,10 +17,12 @@
 
 
 Name:           libyui-ncurses-pkg
-Version:        2.49.0
+Version:        2.49.1
 Release:        0
 Source:         %{name}-%{version}.tar.bz2
 
+# NOTE: when increasing the so version add also a new Obsoletes dependency
+# at [*] below
```

Comment:
We could avoid this with a bit of [macro ~magic~ advanced technology](https://rpm.org/user_doc/macros.html):
```
%define previous_so_version %(expr %{so_version} - 1)
```

Reply:
Well, that might not be enough, we need to obsolete *all* older versions, imagine:

- SLE15-SP1: libyui9
- SLE15-SP2: libyui10
- SLE15-SP3: libyui11

That would work fine when upgrading from SP1 to SP2 or from SP2 to SP3. But for SP1 to SP3 upgrade it would not help, we would need to obsolete also the `so_version - 2` version.

We could do:

```
%define so_version_minus1 %(expr %{so_version} - 1)
%define so_version_minus2 %(expr %{so_version} - 2)
%define so_version_minus3 %(expr %{so_version} - 3)
%define so_version_minus4 %(expr %{so_version} - 4)

Obsoletes: libyui-ncurses-pkg%{so_version_minus1}
Obsoletes: libyui-ncurses-pkg%{so_version_minus2}
Obsoletes: libyui-ncurses-pkg%{so_version_minus3}
Obsoletes: libyui-ncurses-pkg%{so_version_minus4}
```

That could save some work and avoid forgetting a new value, but that still might not work as expected. We do not know how many previous versions we will need, it's not guaranteed that 4 will be enough during the product lifetime.

I was already thinking about some magic but I haven't found any nice solution for this...  Is it possible to define some loop with `[1..%{so_version} - 1]` range in *.spec so we could automatically obsolete all versions down to 1? :thinking:

Any idea?

Reply:
I don't know if the spec engine handles multi line expansions, but watch this:

```console
$ rpm --eval '%(printf "Obsoletes: libyui-ncurses-pkg%d\n" $(seq 4))'
Obsoletes: libyui-ncurses-pkg1
Obsoletes: libyui-ncurses-pkg2
Obsoletes: libyui-ncurses-pkg3
Obsoletes: libyui-ncurses-pkg4
```